File: Medium/1503. Last Moment Before All Ants Fall Out of a Plank/1503. Last Moment Before All Ants Fall Out of a Plank.py
from typing import List


class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        left_time = max(left) if left else 0
        right_time = max(n - position for position in right) if right else 0
        return max(left_time, right_time)

File: Medium/1503. Last Moment Before All Ants Fall Out of a Plank/1503. Last Moment Before All Ants Fall Out of a Plank.rs
impl Solution {
    pub fn get_last_moment(n: i32, left: Vec<i32>, right: Vec<i32>) -> i32 {
        let left_time = left.iter().map(|&position| position).max().unwrap_or(0);
        let right_time = right
            .iter()
            .map(|&position| n - position)
            .max()
            .unwrap_or(0);
        std::cmp::max(left_time, right_time)
    }
}
